pkg/queryfrontend/shard_query.go

```diff
@@ -55,11 +55,8 @@ type querySharder struct {
 
 func (s querySharder) Do(ctx context.Context, r queryrange.Request) (queryrange.Response, error) {
 	analysis, err := s.queryAnalyzer.Analyze(r.GetQuery())
```

Comment:
I am wondering which case it will return error. @sdufel Can you provide your queries?

Reply:
It seems to be an issue in cases of mismatched parentheses or curly braces.  I don't have an exhaustive list, but Imagine any query which isn't syntactically correct triggers this issue.
For instance `foo(` is a minimal query with which you can see the problem.

Reply:
I see. Thanks for the explanation.